test/unit/LevelPlaneTest.js

```diff
@@ -433,3 +433,69 @@ test('torch charge: place block', t => {
 
   t.end();
 });
+
+test('piston activate: place block', t => {
+  const data = [
+    '','','grass','pistonLeft','','pistonRight','grass','',
+    '','','','redstoneWireVertical','','redstoneWireVertical','','',
+    '','','grass','redstoneWireVertical','','redstoneWireVertical','','',
+    '','','pistonUp','redstoneWireUpLeft','','redstoneWireUpRight','pistonDown','',
+    '','','','','','','grass','',
+    '','','','','','','','',
+  ];
+  const plane = new LevelPlane(data, 8, 6, true);
+
+  plane.setBlockAt([4, 2], new LevelBlock('railsRedstoneTorch'));
+
+  const expected = [
+    '','grass','pistonArm','pistonLeft','','pistonRight','pistonArm','grass',
+    '','','grass','redstoneWireVerticalOn','','redstoneWireVerticalOn','','',
+    '','','pistonArm','redstoneWireTRightOn','railsRedstoneTorch','redstoneWireTLeftOn','','',
+    '','','pistonUp','redstoneWireUpLeftOn','','redstoneWireUpRightOn','pistonDown','',
+    '','','','','','','pistonArm','',
+    '','','','','','','grass','',
+  ];
+
+  expected.width = undefined;
+  expected.height = undefined;
+  expected.levelModel = null;
+  expected.redstoneList = [];
+  expected.redstoneListON = [];
```

Comment:
Why is the `redstoneListON` empty here?

Reply:
Because I'm not evaluating the redstone wires themselves. Just the piston states. That list would have all the wire that's powered, which we could evaluate, but as long as the Pistons are getting their state updated correctly, that's all I was concerned with.